lib/Sema/DerivedConformanceDistributedActor.cpp

```diff
@@ -477,7 +476,22 @@ static ValueDecl *deriveDistributedActor_actorSystem(
   propDecl->getAttrs().add(
       new (C) NonisolatedAttr(/*IsImplicit=*/true));
 
-  derived.addMembersToConformanceContext({ propDecl, pbDecl });
+  // IMPORTANT: `id` MUST be the first field of a distributed actor, and
+  // `actorSystem` MUST be the second field, because for a remote instance
+  // we don't allocate memory after those two fields, so their order is very
+  // important. The `hint` below makes sure the system is inserted right after.
+  if (auto id = derived.Nominal->getDistributedActorIDProperty()) {
+    derived.addMemberToConformanceContext(pbDecl, /*hint=*/id);
+    derived.addMemberToConformanceContext(propDecl, /*hint=*/id);
+  } else {
```

Comment:
Would this `else` ever be executed? Isn't `getDistributedActorIDProperty` supposed to always return a valid id field either by returning pre-existing or synthesizing one?

Reply:
Hm yeah you’re right tbh.. I’ll update that

Reply:
Hm actually the nullptr might be returned when a cycle is detected in the request evaluator... Perhaps better safe and sorry here?

Reply:
Double checked -- you're right :) id does not request system so we're good